cpp/src/ray/runtime/abstract_ray_runtime.h

```diff
@@ -88,7 +88,7 @@ class AbstractRayRuntime : public RayRuntime {
 
   const TaskID &GetCurrentTaskId();
 
-  const JobID &GetCurrentJobID();
+  JobID GetCurrentJobID();
```

Comment:
Why update this one but now the rest?

Reply:
this calls to the context which returns by value instead of reference; so we change it accordingly.
change them to return by value is a great idea, but will yield double the size of the PR and touching a lot of cpp runtime code, thus we prefer not changing it in this PR.